tests/foreman/cli/test_leapp_client.py

```diff
@@ -0,0 +1,344 @@
+    # https://docs.pytest.org/en/latest/reference/reference.html#stash
+    request.node.stash[synced_repos] = {}
+    yield request.node.stash
+
+
+RHEL7_VER = '7.9'
+RHEL8_VER = '8.8'
+RHEL9_VER = '9.2'
+
+RHEL_REPOS = {
+    'rhel7_server': {
+        'id': 'rhel-7-server-rpms',
+        'name': f'Red Hat Enterprise Linux 7 Server RPMs x86_64 {RHEL7_VER}',
+        'releasever': RHEL7_VER,
+        'reposet': 'Red Hat Enterprise Linux 7 Server (RPMs)',
+        'product': 'Red Hat Enterprise Linux Server',
+    },
+    'rhel7_server_extras': {
+        'id': 'rhel-7-server-extras-rpms',
+        'name': 'Red Hat Enterprise Linux 7 Server - Extras RPMs x86_64',
+        'releasever': '7',
+        'reposet': 'Red Hat Enterprise Linux 7 Server - Extras (RPMs)',
+        'product': 'Red Hat Enterprise Linux Server',
+    },
+    'rhel8_bos': {
+        'id': 'rhel-8-for-x86_64-baseos-rpms',
+        'name': f'Red Hat Enterprise Linux 8 for x86_64 - BaseOS RPMs {RHEL8_VER}',
+        'releasever': RHEL8_VER,
+        'reposet': 'Red Hat Enterprise Linux 8 for x86_64 - BaseOS (RPMs)',
+    },
+    'rhel8_aps': {
+        'id': 'rhel-8-for-x86_64-appstream-rpms',
+        'name': f'Red Hat Enterprise Linux 8 for x86_64 - AppStream RPMs {RHEL8_VER}',
+        'releasever': RHEL8_VER,
+        'reposet': 'Red Hat Enterprise Linux 8 for x86_64 - AppStream (RPMs)',
+    },
+    'rhel9_bos': {
+        'id': 'rhel-9-for-x86_64-baseos-rpms',
+        'name': f'Red Hat Enterprise Linux 9 for x86_64 - BaseOS RPMs {RHEL9_VER}',
+        'releasever': RHEL9_VER,
+        'reposet': 'Red Hat Enterprise Linux 9 for x86_64 - BaseOS (RPMs)',
+    },
+    'rhel9_aps': {
+        'id': 'rhel-9-for-x86_64-appstream-rpms',
+        'name': f'Red Hat Enterprise Linux 9 for x86_64 - AppStream RPMs {RHEL9_VER}',
+        'releasever': RHEL9_VER,
+        'reposet': 'Red Hat Enterprise Linux 9 for x86_64 - AppStream (RPMs)',
+    },
+}
+
+
+@pytest.fixture
+def function_leapp_cv(module_target_sat, module_sca_manifest_org):
+    logger.info('Creating Leapp Content View')
+    return module_target_sat.api.ContentView(organization=module_sca_manifest_org).create()
+
+
+@pytest.fixture(scope='module')
+def module_leapp_lce(module_target_sat, module_sca_manifest_org):
+    logger.info('Creating Leapp Lifecycle Environment')
+    return module_target_sat.api.LifecycleEnvironment(organization=module_sca_manifest_org).create()
+
+
+@pytest.fixture
+def function_leapp_ak(
+    module_target_sat, function_leapp_cv, module_leapp_lce, module_sca_manifest_org, upgrade_path
+):
+    logger.info(
+        'Creating Leapp Activation Key for RHEL %s -> %s',
+        upgrade_path['source_version'],
+        upgrade_path['target_version'],
+    )
+    return module_target_sat.api.ActivationKey(
+        content_view=function_leapp_cv,
+        environment=module_leapp_lce,
+        organization=module_sca_manifest_org,
+    ).create()
```

Comment:
Just to optimize the count of fixture in use, can we move this LCE/CV/AK create fixtures into a `leapp_sat_content` fixture, as this fixtures be used mostly there and not in test directly?
I think AK will be used that too just for registration so it can be returned from `leapp_sat_content` fixture as well, as it returns nothing atm, wdyt?

Reply:
I understand your goal of optimizing the number of fixtures in use, and I agree that this could be a good idea in the long run.

However, I think it is important to get the PR merged ASAP, and making the changes you suggest would take some time to change and test. I also think the current structure is more granular and is logically split into individual fixtures. This makes it easier to understand and maintain the code.

[I have also commented](https://github.com/SatelliteQE/robottelo/pull/11553#discussion_r1318867683) on the leapp_sat_content fixture with some potential changes that could make it more efficient. I think these changes would be a good compromise between optimizing the number of fixtures and keeping the code easy to understand and maintain.

Reply:
Based upon the @ogajduse [comment](https://github.com/SatelliteQE/robottelo/pull/11553/commits/d6a87722820f60b675d2d9f8602d1eeea053a955#r1262816633) and suggested nice fixture example, all above small tasks breakdown (converted from function) into reusable fixture by considering any new test case would use the fixtures, I could see you have voted for that.